File: src/main/frontend/src/models/uiRequest.ts
export interface UiRequest {
	gameId: number;
	type: UiRequestType;
	params: number[];
}

export enum UiRequestType {
	AUCTION,
	BUY,
	DO_NOT_BUY,
	DOWNGRADE,
	END_TURN,
	JAIL_GOJF,
	JAIL_ROLL,
	JAIL_PAY,
	JAIL_WAIT,
	ROLL,
	SELL,
	TRADE,
	UNMORTGAGE,
	UPGRADE,
}

File: src/main/java/com/smooth038/monopoly/game/UiRequest.java
package com.smooth038.monopoly.game;

public class UiRequest {
    private int gameId;
    private UiRequestType type;
    private int[] params;

    public UiRequest(int gameId, UiRequestType type, int[] params) {
        this.gameId = gameId;
        this.type = type;
        this.params = params;
    }

    public int getGameId() {
        return gameId;
    }

    public UiRequestType getType() {
        return type;
    }

    public int[] getParams() {
        return params;
    }
}
